src/test/kotlin/uk/co/craigbass/pratura/unit/AddProductSpec.kt

```diff
@@ -9,24 +9,32 @@ import uk.co.craigbass.pratura.testdouble.InMemoryProductGateway
 import uk.co.craigbass.pratura.usecase.AddProduct
 
 class AddProductSpec : Spek({
-
   val productGateway = InMemoryProductGateway()
-
+  val useCase = memoized { AddProduct(productGateway) }
   beforeEachTest {
-    AddProduct(productGateway).execute(Request("sku:1029317", "1.00".toDecimal()))
+    useCase().execute(Request(
+      sku = "sku:1029317",
+      price = "6.99".toDecimal(),
+      name = "Pack of 4 AA Batteries"
+    ))
   }
 
+  val savedProducts = memoized { productGateway.all() }
+  val firstSavedProduct = memoized { savedProducts().first() }
+
   it("should have saved one product") {
```

Comment:
I would combine these assertion blocks into one e.g. "It should have saved the correct product information" then 3 assertion lines. Having all this boiler plate for 1 assertion line is equivalent to commenting every line. If your code is readable then there should be no harm in combining them. It is one logical assertion after all.